src/analysis/calculus/inverse.lean

```diff
@@ -496,3 +496,90 @@ theorem to_local_left_inverse {g : 𝕜 → 𝕜} (hg : ∀ᶠ x in 𝓝 a, g (f
 (hf.has_strict_fderiv_at_equiv hf').to_local_left_inverse hg
 
 end has_strict_deriv_at
+
+/-!
+### Inverse function theorem, smooth case
+
+-/
+
+namespace times_cont_diff_at
+variables {E' : Type*} [normed_group E'] [normed_space ℝ E']
+variables {F' : Type*} [normed_group F'] [normed_space ℝ F']
+variables [complete_space E'] (f : E' → F') {f' : E' ≃L[ℝ] F'} {a : E'}
+
+/-- Given a `times_cont_diff` function over `ℝ` with an invertible derivative at `a`, returns a
+`local_homeomorph` with `to_fun = f` and `a ∈ source`. -/
+def to_local_homeomorph
+  {n : with_top ℕ} (hf : times_cont_diff_at ℝ n f a) (hf' : has_fderiv_at f (f' : E' →L[ℝ] F') a)
+  (hn : 1 ≤ n) :
+  local_homeomorph E' F' :=
+begin
+  have hf₂ := hf.has_strict_fderiv_at hn,
+  rw hf'.fderiv at hf₂,
+  exact hf₂.to_local_homeomorph f
+end
+
+variable {f}
+
+@[simp] lemma to_local_homeomorph_coe
+  {n : with_top ℕ} (hf : times_cont_diff_at ℝ n f a) (hf' : has_fderiv_at f (f' : E' →L[ℝ] F') a)
+  (hn : 1 ≤ n) :
+  (hf.to_local_homeomorph f hf' hn : E' → F') = f := rfl
+
+lemma mem_to_local_homeomorph_source
+  {n : with_top ℕ} (hf : times_cont_diff_at ℝ n f a) (hf' : has_fderiv_at f (f' : E' →L[ℝ] F') a)
+  (hn : 1 ≤ n) :
+  a ∈ (hf.to_local_homeomorph f hf' hn).source :=
+begin
+  have hf₂ := hf.has_strict_fderiv_at hn,
+  rw hf'.fderiv at hf₂,
+  exact hf₂.mem_to_local_homeomorph_source
+end
+
+lemma image_mem_to_local_homeomorph_target
+  {n : with_top ℕ} (hf : times_cont_diff_at ℝ n f a) (hf' : has_fderiv_at f (f' : E' →L[ℝ] F') a)
+  (hn : 1 ≤ n) :
+  f a ∈ (hf.to_local_homeomorph f hf' hn).target :=
+begin
+  have hf₂ := hf.has_strict_fderiv_at hn,
+  rw hf'.fderiv at hf₂,
+  exact hf₂.image_mem_to_local_homeomorph_target
+end
+
+lemma image_to_local_homeomorph_image_basepoint
```

Comment:
With the change proposed above you should be able to prove it using `(hf.has_strict_fderiv_at' hf' hn).local_inverse_apply_image`. Please sync names with `has_strict_fderiv_at`.